cpp/dolfin/fem/DirichletBC.h

```diff
@@ -42,7 +42,7 @@ namespace fem
 /// The boundary indicators may be specified in a number of different
 /// ways:
 ///
-/// 1. Providing a_SubDomain_ object, using the inside() function to
+/// 1. Providing a marking function, mark(x, only_boundary), to
```

Comment:
`on_boundary` make more sense.

Reply:
This is the issue. The past (and current) implementation means `only_boundary`. Because `false` means all vertices, with `on_boundary` it would mean only interior vertices.